Lib/test/test_pickle.py

```diff
@@ -507,6 +507,41 @@ def test_exceptions(self):
                     self.assertEqual(mapping('exceptions', name),
                                      ('builtins', name))
 
+    def test_custom_exceptions(self):
+        global CustomException1, CustomException2, CustomException3
+        class CustomException1(Exception):
+            def __init__(self,a,b):
+                c = f"{a}{b}"
+                super().__init__(c)
+        class CustomException2(Exception):
+            def __init__(self,a,**b):
+                c = f"{a}{b}"
+                super().__init__(c)
+        class CustomException3(Exception):
+            def __init__(self,a,b):
+                super().__init__()
+                self.c = f"{a}{b}"
+
+        CustomExc1 = CustomException1(1,'2')
+        CustomExc2 = CustomException2(1,test='2')
+        CustomExc3 = CustomException3(1,'2')
+
+        self.assertEqual(str(CustomExc1), "12")
+        self.assertEqual(str(CustomExc2), "1{'test': '2'}")
+        self.assertEqual(str(CustomExc3), "")
+
+        CustomExc1_pickled = pickle.dumps(CustomExc1)
+        CustomExc1_unpickled = pickle.loads(CustomExc1_pickled)
+        self.assertEqual(CustomExc1.args, CustomExc1_unpickled.args)
```

Comment:
Test CustomExc1_unpickled.args against a constant instead, like
```
self.assertEqual(CustomExc1_unpickled.args, ("12",))
```
That makes the test easier to read.